### DISCUSSION TITLE
Non-spinning multi-barrel submarine turrets

### DISCUSSION THREAD

**Discussion by ssparky** [FakeFishGames/Barotrauma#13609] [Ideas and suggestions]
There is currently no good way to mod in a multi-barrel turret where each barrel recoils individually. You can either use a regular "Barrel" part, which will make all barrels recoil simultaneously, or a "SpinningBarrel" part, which **does** support individual barrel recoil, but forces the barrels to spin.
Having the option for multi-barrel turrets where the barrels do not spin would allow for more variety and creativity in modded turret designs.
There are many ways this could be implemented, but the simplest (in my assumption) way would be to allow changing the rotation speed on the "SpinningBarrel" part, specifically with support for 0 rotation speed.

**Comment by ubertpendragon** [comment]
Double coilgun? 🤔

**Reply by ssparky** [reply]
Double coilgun uses the regular Barrel option, meaning both its barrels recoil simultaneously.
